pkg/resources/pool_stub.go

```diff
@@ -50,63 +45,59 @@ func (rp *resourcePool) InitDevice() error {
 	// Not implemented
 	return nil
 }
-func (rp *resourcePool) DiscoverDevices() error {
-	// discover SRIOV VFs
-	// populate devices[] for each device discovered
-	glog.Infof("Discovering devices with config: %+v", rp.config)
-	deviceList := make([]string, 0)
-	SriovMode := rp.config.SriovMode
-
-	if SriovMode {
-
-		// Discover VFs
-		for _, pf := range rp.config.RootDevices {
-			numvfs := utils.GetSriovVFcapacity(pf)
-			glog.Infof("Total number of VFs for device %v is %v", pf, numvfs)
-			if numvfs > 0 {
-				glog.Infof("SRIOV capable device discovered: %v", pf)
-
-				numConfiguredVFs := utils.GetVFconfigured(pf)
-				glog.Infof("Number of Configured VFs for device %v is %d", pf, numConfiguredVFs)
-
-				if numConfiguredVFs < 1 {
-					glog.Errorf("Error. Virtual Functions are not configured for the device: %s", pf)
-					return fmt.Errorf("Virtual Functions are not configured for the device: %s", pf)
-				}
-
-				if newList, err := utils.GetVFList(pf); err == nil {
-					deviceList = append(deviceList, newList...)
-				}
-			}
-		}
-	} else { // PFs only
-		deviceList = rp.config.RootDevices
-	}
-
-	// Create plugin.Device instaces for all devices
-	for _, dev := range deviceList {
-		devFile, err := rp.GetDeviceFile(dev)
-		if err == nil {
-			rp.devices[dev] = &pluginapi.Device{
-				ID:     dev,
-				Health: pluginapi.Healthy}
-			rp.deviceFiles[dev] = devFile
-		}
-	}
-	glog.Infof("Discovered Devices: %v\n", rp.devices)
-
-	return nil
-}
 
 func (rp *resourcePool) GetResourceName() string {
-	return rp.config.ResourceName // dummy return
+	return rp.config.ResourceName
 }
 
 func (rp *resourcePool) GetDevices() map[string]*pluginapi.Device {
 	// returns all devices from devices[]
 	return rp.devices
 }
 
-func (rp *resourcePool) GetDeviceFiles() map[string]string {
-	return rp.deviceFiles
+func (rp *resourcePool) Probe() bool {
+	// TO-DO: Implement this
+	return false
+}
+
+func (rp *resourcePool) GetDeviceSpecs(deviceIDs []string) []*pluginapi.DeviceSpec {
+	glog.Infof("GetDeviceSpecs(): for devices: %v", deviceIDs)
+	devSpecs := make([]*pluginapi.DeviceSpec, 0)
+
+	// Add vfio group specific devices
+	for _, id := range deviceIDs {
+		if dev, ok := rp.devicePool[id]; ok {
+			ds := dev.GetDeviceSpecs()
+			devSpecs = append(devSpecs, ds...)
```

Comment:
For vfio devices `dev.GetDeviceSpecs()` each device spec returned contains its specific mount and one for the default `/dev/vfio/vfio`. When multiple devices are passed here, in the final returned DeviceSpec this default `/dev/vfio/vfio` spec appears multiple times. Can this be an issue?

Reply:
Good point! Functionally this shouldn't be any issue. As this will just add the same device file into the same cgroup again. Ideally we should add a device file once in the overall deviceSpec. We can add a check - only add deviceSpec if its not in the spec list. I will add a fix. Thanks